File: realtime-server/src/main/java/com/slowhigh/kairos/timer/dto/TargetTimeUpdateEvent.java
package com.slowhigh.kairos.timer.dto;

import java.time.Instant;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "'targetTimeUpdate' 이벤트의 데이터 형식")
public class TargetTimeUpdateEvent {

  @Schema(description = "갱신된 타임스탬프", example = "2025-07-08T15:00:00Z")
  private Instant updatedAt;

  @Schema(description = "서버 시간", example = "2025-07-08T15:00:00Z")
  private Instant serverTime;

  @Schema(description = "신규 기준 시간", example = "2025-07-08T15:00:00Z")
  private Instant newTargetTime;

}

File: server/apps/sync-service/src/main/kotlin/com/sharetimer/syncservice/adapter/output/web/dto/TargetTimeUpdateEvent.kt
package com.sharetimer.syncservice.adapter.output.web.dto

import io.swagger.v3.oas.annotations.media.Schema
import java.time.Instant

@Schema(description = "Data format for 'targetTimeUpdate' event")
data class TargetTimeUpdateEvent(
    @field:Schema(description = "Updated timestamp", example = "2025-07-08T15:00:00Z")
    val updatedAt: Instant,
    @field:Schema(description = "Server time", example = "2025-07-08T15:00:00Z")
    val serverTime: Instant,
    @field:Schema(description = "New target time", example = "2025-07-08T15:00:00Z")
    val newTargetTime: Instant,
)
